### ISSUE TITLE
Edge case where arrow-parens incorrectly complains and fixer produces invalid output

### ISSUE THREAD

**Issue by richardm** [palantir/tslint#2752]
### Bug Report

- __TSLint version__: 4.4.2
- __TypeScript version__: 2.1.5
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
angular
  .module('app')
  .filter('myFilter', function () {
    return (items): Items[] => {
      // ...

      return items;
    };
  });
```

with `tslint.json` configuration:

```json
{
  "extends": "tslint:recommended",
  "rules": {
    "arrow-parens": [true, "ban-single-arg-parens"]
  }
}
```

#### Actual behavior
(items) is shown as invalid, and the fixer outputs the following invalid code:

```ts
angular
  .module('app')
  .filter('myFilter', function () {
    return items; : Items[]; => {
      // ...

      return items;
    };
  });
```

#### Expected behavior
This should not error or produce bad output, as the parens are necessary here

**Comment by ajafff** [comment]
That should be fixed in a newer version of tslint. Testing this locally with v5.2.0 works correct.
